### PR TITLE
Setup messaging queue for product service

### LINKED ISSUE #33: Set up CQRS in Review micro-service
**Description**
Makes it possible to process messages et cetera

**Why is this needed**
It won't do anything otherwise.

**Tasks**
- [ ] Set up commands
- [ ] Set up queries

### PR DESCRIPTION
**Pull request by Rubenvdbrink**
(I accidentally messed up the branch feature number)
## Changes
RabbitMq implementation following Alex' example. 

## Checklist
- [x] Add some exclusions for jacoco & sonarcloud.
- [x] Add properties for messaging for easy changeability.
- [x] General messaging queue setup.

## When merging, this pull request 
closes #47

### CODE REVIEW

File: product/src/main/java/nl/softwarestrijders/waiter/product/infrastructure/config/RabbitMqConfig.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    @Bean
+    public Binding productBinding() {
+        return BindingBuilder.bind(productQueue()).to(waiterExchange()).with(this.productRoutingKey);
+    }
+
+    @Bean
+    public ConnectionFactory connectionFactory() {
+        return new CachingConnectionFactory(this.host, this.port);
+    }
```
Comment (JustMilan) [inline product/src/main/java/nl/softwarestrijders/waiter/product/infrastructure/config/RabbitMqConfig.java:45]: Nit: put all returns on a new line or none at all. For consistency.

File: product/src/main/resources/application.properties [hunk 2]
```diff
@@ -4,7 +4,11 @@ server.port=8083
...
 spring.data.mongodb.auto-index-creation=true
-# RabbitMQ
+# Spring & RabbitMq
 spring.rabbitmq.host=localhost
 spring.rabbitmq.port=5672
 spring.rabbitmq.publisher-confirm-type=simple
+# Messaging
+messaging.exchange.waiter=waiter
+messaging.queue.product=product
+messaging.routing-key.product=product.#
```
Comment (JustMilan) [inline product/src/main/resources/application.properties:14]: Please insert a new line at the eof

**Review by JustMilan — CHANGES_REQUESTED** [review]

File: product/src/main/java/nl/softwarestrijders/waiter/product/infrastructure/config/RabbitMqConfig.java (hunk 1 above)
Comment (xandervedder) [reply product/src/main/java/nl/softwarestrijders/waiter/product/infrastructure/config/RabbitMqConfig.java:45]: Agreed

File: product/src/main/resources/application.properties (hunk 2 above)
Comment (xandervedder) [reply product/src/main/resources/application.properties:14]: Agreed

**Review by JustMilan — APPROVED** [review]

**Review by xandervedder — APPROVED** [review]

**Pull request merged**